bundles/sirix-core/src/main/java/org/sirix/service/xml/shredder/WikipediaImport.java

```diff
@@ -276,31 +276,33 @@ private void updateShredder() throws SirixException, IOException, XMLStreamExcep
     final DatabaseConfiguration dbConf = new DatabaseConfiguration(path);
     Databases.removeDatabase(path);
     Databases.createXmlDatabase(dbConf);
-    final var db = Databases.openXmlDatabase(path);
-    db.createResource(new ResourceConfiguration.Builder("wiki").build());
-    final XmlResourceManager resourceManager = db.openResourceManager("wiki");
-    if (mPageEvents.peek().isStartElement()
-        && !mPageEvents.peek().asStartElement().getName().getLocalPart().equals("root")) {
-      mPageEvents.addFirst(XMLEventFactory.newInstance().createStartElement(new QName("root"), null, null));
-      mPageEvents.addLast(XMLEventFactory.newInstance().createEndElement(new QName("root"), null));
-    }
-    final XmlNodeTrx wtx = resourceManager.beginNodeTrx();
-    final XmlShredder shredder = new XmlShredder.Builder(wtx, XmlShredder.createQueueReader(mPageEvents),
-        InsertPosition.AS_FIRST_CHILD).commitAfterwards().build();
-    shredder.call();
-    wtx.close();
-    mPageEvents = new ArrayDeque<>();
-    final XmlNodeReadOnlyTrx rtx = resourceManager.beginNodeReadOnlyTrx();
-    rtx.moveToFirstChild();
-    rtx.moveToFirstChild();
-    final long nodeKey = mWtx.getNodeKey();
-    try (final FMSE fmse = FMSE.createInstance(new DefaultNodeComparisonFactory())) {
-      fmse.diff(mWtx, rtx);
+    try (var db = Databases.openXmlDatabase(path)) {
+      db.createResource(new ResourceConfiguration.Builder("wiki").build());
+      try (XmlResourceManager resourceManager = db.openResourceManager("wiki")) {
+        if (mPageEvents.peek() != null && mPageEvents.peek().isStartElement()
+                && !mPageEvents.peek().asStartElement().getName().getLocalPart().equals("root")) {
+          mPageEvents.addFirst(XMLEventFactory.newInstance().createStartElement(new QName("root"), null, null));
+          mPageEvents.addLast(XMLEventFactory.newInstance().createEndElement(new QName("root"), null));
+        }
+        final XmlNodeTrx wtx = resourceManager.beginNodeTrx();
+        final XmlShredder shredder = new XmlShredder.Builder(wtx, XmlShredder.createQueueReader(mPageEvents),
+                InsertPosition.AS_FIRST_CHILD).commitAfterwards().build();
+        shredder.call();
+        wtx.close();
+        mPageEvents = new ArrayDeque<>();
+        final XmlNodeReadOnlyTrx rtx = resourceManager.beginNodeReadOnlyTrx();
+        rtx.moveToFirstChild();
+        rtx.moveToFirstChild();
+        final long nodeKey = mWtx.getNodeKey();
+        try (final FMSE fmse = FMSE.createInstance(new DefaultNodeComparisonFactory())) {
+          fmse.diff(mWtx, rtx);
+        }
+        mWtx.moveTo(nodeKey);
+        rtx.close();
+        resourceManager.close();
+      }
+      db.close();
```

Comment:
Shouldn't we remove all these `.close()` as the resources are automatically closed by `try`?

Reply:
True, after the change it's implicitly closed, even probably before (I'd have to check later on after work, what the FMSE close() method does.